src/ucp/core/ucp_request.c

```diff
@@ -317,8 +317,14 @@ int ucp_request_pending_add(ucp_request_t *req)
 static unsigned ucp_request_dt_invalidate_progress(void *arg)
 {
     ucp_request_t *req = arg;
+    ucp_ep_h ucp_ep    = req->send.ep;
 
     ucp_request_complete_send(req, req->status);
+
+    /* Now UCP endpoint could be destoyed if it was asked */
+    UCP_EP_ASSERT_COUNTER_DEC(&ucp_ep->invalidate_req_refcount);
+    ucp_ep_remove_ref(ucp_ep);
```

Comment:
can we do some macro/helper like
"ucp_ep_add_ref(ep, kind)" where kind is the type for debug purpose only (like invalidate, discard, etc)
to force specifying the kind when doing add/del ref

Reply:
done